### ISSUE TITLE
`UnnecessaryQualifier` outputs incorrect warnings?

### ISSUE THREAD

**Issue by facboy** [google/error-prone#5129]
actually before looking at #5126, #5127, and #5128, is `UnnecessaryQualifier` even working correctly at all?

given

```
class MyClass {

  ...

  public MyClass(@MyQualifier Some thing) {
  }
}
```

```
@Qualifier
@interface MyQualifier {}
```

it's complaining about the qualifier on the constructor parameter, which is basically the exact example of valid usage in the doc.

**Comment by tbroyer** [comment]
Your constructor is not annotated with `@Inject` or similar so it's an "injection point" and indeed the qualifier annotation is unnecessary here.

**Comment by facboy** [comment]
bleh, right you are! i probably removed it at some point during a refactor.
